jwst/emicorr/emicorr.py

```diff
@@ -438,25 +438,23 @@ def apply_emicorr(output_model, emicorr_model,
         # This is the phase matched noise model to subtract from each pixel of the input image
         dd_noise = lut[(phaseall * period_in_pixels).astype(int)]
 
-        # Interleave (straight copy) into 4 amps
-        noise = np.zeros((nints, ngroups, ny, nx))   # same size as input data
-        noise_x = np.arange(nx4) * 4
-        for k in range(4):
-            noise[:, :, :, noise_x + k] = dd_noise
-
         # Safety catch; anywhere the noise value is not finite, set it to zero
-        noise[~np.isfinite(noise)] = 0.0
+        dd_noise[~np.isfinite(dd_noise)] = 0.0
 
         # Subtract EMI noise from the input data
         log.info('Subtracting EMI noise from data')
-        output_model.data = output_model.data - noise
+
+        # Interleave (straight copy) into 4 amps
+        noise_x = np.arange(nx4) * 4
+        for k in range(4):
+            output_model.data[..., noise_x + k] = output_model.data[..., noise_x + k] - dd_noise
```

Comment:
This might avoid an additional copy of the output_model.data:

```suggestion
            output_model.data[..., noise_x + k] -= dd_noise
```

Reply:
I tried this, and though the step runs nicely, the tests crash due to an incompatibility of ints and floats, and for some reason this is not solved by forcing the array to be of a specific type

Reply:
It looks like that's because the tests use a Level1bModel instead of a RampModel.  Since the input for the emicorr step is expected to be the output from dq_init, the RampModel is probably more appropriate.

Reply:
Thanks, it worked!